valhalla/baldr/graphtile.h

```diff
@@ -221,16 +231,25 @@ class GraphTile {
                              " transitioncount= " + std::to_string(header_->transitioncount()));
   }
 
+  /**
+   * Get an iterable set of transitions from a node in this tile
+   * @param  node  Node from which the transitions leave
+   * @return returns an iterable collection of node transitions
+   */
+  midgard::iterable_t<const NodeTransition> GetNodeTransitions(const NodeInfo* node) const {
+    const auto* trans = transitions_ + node->transition_index();
+    return midgard::iterable_t<const NodeTransition>{trans, node->transition_count()};
+  }
+
   /**
    * Get an iterable set of transitions from a node in this tile
    * @param  node  GraphId of the node from which the transitions leave
    * @return returns an iterable collection of node transitions
    */
   midgard::iterable_t<const NodeTransition> GetNodeTransitions(const GraphId& node) const {
     if (node.id() < header_->nodecount()) {
-      const auto& nodeinfo = nodes_[node.id()];
-      const auto* trans = transition(nodeinfo.transition_index());
-      return midgard::iterable_t<const NodeTransition>{trans, nodeinfo.transition_count()};
+      const auto* nodeinfo = nodes_ + node.id();
+      return GetNodeTransitions(nodeinfo);
```

Comment:
similar to the other helper i added so that we can use the node rather than the id